lib/school_of_fish.rb

```diff
@@ -24,22 +35,88 @@ def update(dt)
     elsif (rand<0.01)
       @y_fish_speed = 1*rand-0.5
     end
+
     @x_position += @x_fish_speed
     @y_position += @y_fish_speed
     @rotation += dt*@rotation_speed
+
+    disperse(dt)
```

Comment:
after it has completed dispersion, this instance does not need to be updated anymore. Also, references to this instance (inside the window) should be eliminated so that it doesn't waste resources